contracts/axelar-interchain-token-service/src/contract.rs

```diff
@@ -0,0 +1,100 @@
+use axelar_soroban_std::ensure;
+use soroban_sdk::{contract, contractimpl, Address, Env, String,};
+
+use crate::error::InterchainTokenServiceError;
+use crate::event;
+use crate::storage_types::{DataKey, TrustedItsAddress};
+
+#[contract]
+pub struct InterchainTokenService;
+
+#[contractimpl]
+impl InterchainTokenService {
+    pub fn initialize(
+        env: Env,
+        owner: Address
+    ) -> Result<(), InterchainTokenServiceError> {
+        ensure!(
+            env.storage()
+                .instance()
+                .get(&DataKey::Initialized)
+                .unwrap_or(true),
+            InterchainTokenServiceError::AlreadyInitialized
+        );
+
+        env.storage().instance().set(&DataKey::Initialized, &true);
+
+        env.storage().instance().set(&DataKey::Owner, &owner);
+
+        Ok(())
+    }
+
+    pub fn owner(env: &Env) -> Address {
+        env.storage().instance().get(&DataKey::Owner).unwrap()
```

Comment:
I'm fine with an unwrap here instead of a result type since it can only happen due to incomplete initialization.

@hydrobeam Does stellar team have a recommendation? Also, Stellar doesn't distinguish txs from queries, so do we need to expose this as an endpoint in our contracts? Can we query this info from storage of the contract directly in an easy to use manner or do we need to expose an entrypoint? Maybe you can find more info here?

Reply:
My assumption here is is that explicit interaction through entrypoints is desirable. I haven't found any info contrary to this in the docs. I can reach out to Stellar team